libvast/include/vast/error.hpp

```diff
@@ -96,6 +97,21 @@ const char* to_string(ec x);
 /// @relates ec
 std::string render(caf::error err);
 
+/// An error factory that delegates message creation to fmt::format.
+/// @relates ec
+#if FMT_VERSION >= 80000
+template <class... Ts>
+caf::error make_error(ec code, fmt::format_string<Ts...> fmt, Ts&&... args) {
+  return caf::make_error(code, fmt::format(std::move(fmt),
+                                           std::forward<Ts&&>(args)...));
+}
+#else
+template <class S, class... Ts>
+caf::error make_error(ec code, const S& fmt, Ts&&... args) {
+  return caf::make_error(code, fmt::format(fmt, std::forward<Ts&&>(args)...));
+}
+#endif
```

Comment:
Would you mind going one step further and actually poisoning `caf::make_error` when called with `vast::ec` as a first argument type?

```c++
namespace caf {

/// Poison `caf::make_error` for `vast::ec` to ensure users go through the
/// properly formatted version `vast::make_error`.
template <class... Args>
error make_error(::vast::ec, Args&&...) = delete;

} // namespace caf
```

I really like the idea of this, but I think it'd make sense to just do this once and for all so we don't end up with a mixed style in the code base.

Reply:
Thinking about this some more, I think we definitely must do this before we can merge this PR. It's a very bad DX to have two make_error functions with such different behavior.